lib/internal/Magento/Framework/App/MaintenanceMode.php

```diff
@@ -45,36 +47,59 @@ class MaintenanceMode
     private $eventManager;
 
     /**
-     * @param \Magento\Framework\Filesystem $filesystem
+     * @param Filesystem $filesystem
+     * @param Utility\IPAddress $ipAddress
      * @param Manager|null $eventManager
      */
-    public function __construct(Filesystem $filesystem, ?Manager $eventManager = null)
-    {
+    public function __construct(
+        Filesystem $filesystem,
+        private readonly Utility\IPAddress $ipAddress,
+        Manager $eventManager = null,
```

Comment:
Yes. The value should either be supplied with the correct type, or not supplied. By specifying a default value of `= null`, the type implicitly accepts a null value. This is a documentation improvement, and not a code/policy change.

Reply:
Implicit nullable is deprecated in php8.4, so this change should be reverted

Reply:
@thomas-kl1 Yes. That change to PHP happened after this code was authored. When Magento adds support for PHP v8.4+, this will be one of the (likely many) changes required.